srcs/input_handling/input_read.c

```diff
@@ -78,15 +78,21 @@ static int	input_parse(t_vshdata *data)
 		return (IR_EOF);
 	else if (ret == FUNCT_SUCCESS)
 		return (FUNCT_SUCCESS);
-	else if (input_read_ansi(data) == FUNCT_FAILURE)
+	else
 	{
-		if (input_parse_special(data) == FUNCT_FAILURE)
+		ret = input_read_ansi(data);
+		if (ret == FUNCT_FAILURE)
 		{
-			if (input_parse_char(data) == FUNCT_ERROR)
-				return (reset_input_read_return(data, FUNCT_ERROR));
+			if (input_parse_special(data) == FUNCT_FAILURE)
+			{
+				if (input_parse_char(data) == FUNCT_ERROR)
+					return (reset_input_read_return(data, FUNCT_ERROR));
+			}
 		}
+		else if (ret == FUNCT_ERROR)
```

Comment:
This won't trigger if `input_parse_special` returns FUNCT_ERROR, is that correct?